docs/maintainer-guide.md

```diff
@@ -147,6 +148,27 @@ Security vulnerabilities, once reported and confirmed, should be treated as a ca
 Since the detail of such vulnerability cannot be disclosed until it is fixed, an issue can be created just before a PR for the fix is submitted, with minimal information (e.g. simply "Security vulnerability" as an issue with no further description).
 The complete details can be filled in just before merging and/or after the fix is deployed.
 
+### Data migration
+
+Data migration is necessary when changes that are incompatible with the prevailing data storage schema (afterwards "breaking changes") are introduced.
+
+A data migration is initiated by the developer working on the breaking changes and will be handed over to the core team once both the breaking changes and the data migration script(s) are merged.
+
+**Role: RL**
+
+* Release and deploy the new version containing the breaking changes following the normal release workflow. It may be a normal release or a hot patch release.
+* Create an issue to run the data migration script.
+
+**Role: PM**
+
+* (Optional but recommended) Wait for some time to ascertain that the system is stable under the new data schema.
+* Run the data migration script on the live site.
+* Close the data migration issue.
+
+**Role: Any contributor/developer**
+
+* Remove the code that is specifically tailored for the old data schema.
```

Comment:
Comment by @damithc:

Perhaps we should recommend the dev to create issues for all steps of the migration at the beginning so as to ensure the whole process is completed eventually?

Reply:
Comment by @xpdavid:

I think that would be good. Like a issue template?

In doing so, we can know where are we in the data migration.

Sample template below:

This data migration is required because of #

The following tasks has been done:
- [ ] Written code that work for both the old and new versions of the data schema in PR #
- [ ] Created a client script to migrate all data following the old schema to the new schema in PR #.
- [ ] Ran script against LIVE server in issue #
- [ ] Removed the code that is specifically tailored for the old data schema in PR #